enhancements/microshift/periodic-etcd-defragmentation-in-microshift.md

```diff
@@ -0,0 +1,168 @@
+---
+title: periodic-etcd-defragmentation-in-microshift
+authors:
+  - dusk125
+reviewers:
+  - "@hasbro17, etcd Team"
+  - "@tjungblu, etcd Team"
+  - "@Elbehery, etcd Team"
+  - "@fzdarsky"
+  - "@deads2k"
+  - "@derekwaynecarr"
+  - "@mangelajo"
+  - "@pmtk"
+approvers:
+  - "@dhellmann"
+api-approvers:
+  - None
+creation-date: 2023-02-15
+last-updated: 2023-02-15
+tracking-link:
+  - https://issues.redhat.com/browse/ETCD-391
+---
+
+# Periodic etcd Defragmentation in MicroShift
+
+## Summary
+
+This enhancement proposes adding a control loop to automatically and periodically run the etcd defragment commands.
+
+## Motivation
+
+Similar to a spinning-platter harddrive, etcd, through subsequent object creates and deletes, will have its internal database become fragmented. This causes etcd to use more space in memory and on disk than it actually has "in-use".
+Periodic defragmentation (along with compacation invoked by the API Server) is the way to ensure that etcd does not run out of space (hit the maximum-allowed database size of 8GB).
+
+Etcd running in the Openshift Container Platform (OCP) is automatically and periodically defragemented by the Cluster Etcd Operator (CEO). Because operators in Microshift are to be avoided, an alternative control loop for etcd defragmentation needs to be selected.
+
+### User Stories
+
+1. "As a Microshift Device Administrator, I want etcd to automatically defragment itself so that it doesn't run out of space, or use more space than it needs."
+
+### Goals
+
+- Etcd is automatically and periodically defragmented by a control loop that fits into the Microshift paradigm.
+
+### Non-Goals
+
+- Replicate other aspects/control loops of the OCP CEO.
+
+## Proposal
```

Comment:
@dusk125 it would be good to update this section to cover what was actually implemented, and move the alternatives down to the end of the file for future reference